share/transformers/mods.py

```diff
@@ -454,18 +464,10 @@ class RootParser(MODSCreativeWork):
 
         return RootParser
 
-    def do_transform(self, data):
-        approved_sets = self.kwargs.get('approved_sets')
-        if approved_sets is not None:
-            specs = set(x.replace('publication:', '') for x in etree.fromstring(data).xpath(
-                'ns0:header/ns0:setSpec/node()',
-                namespaces={'ns0': 'http://www.openarchives.org/OAI/2.0/'}
-            ))
-            if not (specs & set(approved_sets)):
-                logger.warning('Series %s not found in approved_sets for %s', specs, self.config.label)
-                return (None, None)
-
-        return super().do_transform(data)
+    def do_transform(self, datum):
+        if not oai_allowed_by_sets(datum, self.kwargs.get('blocked_sets'), self.kwargs.get('approved_sets')):
+            return (None, None)
```

Comment:
can we add a logger warning/message here

Reply:
`oai_allowed_by_sets` logs a warning when it returns `False`, I don't think it'd be different in each transformer.